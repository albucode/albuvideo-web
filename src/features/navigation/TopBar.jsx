import React from "react";
import styled from "@emotion/styled";
import { Flex, Text, IconButton } from "@chakra-ui/react";

import { SearchBar } from "../shared/SearchBar";
import Bell from "../shared/icons/Bell";
import Gear from "../shared/icons/Gear";

export const TopBar = ({ sectionName }) => {
  return (
    <Flex>
      <SectionName>{sectionName}</SectionName>
      <SearchBar />
      <ButtonIcon icon={<Bell />} />
      <ButtonIcon icon={<Gear />} />
    </Flex>
  );
};

const SectionName = styled(Text)`
  color: ${(props) => props.theme.colors.black};
  width: 345px;
  font-weight: 700;
  font-size: 36px;
  line-height: 49px;
`;

const ButtonIcon = styled(IconButton)`
  color: ${(props) => props.theme.colors.grey1};
  height: 60px;
  width: 60px;
  background-color: white;
`;
