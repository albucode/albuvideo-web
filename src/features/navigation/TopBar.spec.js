import React from "react";
import { render, screen } from "@testing-library/react";
import { TopBar } from "./TopBar";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import userReducer from "../authentication/userSlice";

const fakeRenderComponent = () => {
  const store = configureStore({
    reducer: {
      user: userReducer,
    },
    preloadedState: {
      user: {
        email: "test@test.com",
        emailInitial: "T",
        isLoggedIn: true,
      },
    },
  });

  const { container } = render(
    <Provider store={store}>
      <TopBar sectionName={"Test"} />
    </Provider>
  );
  return { container };
};

describe("TopBar", () => {
  beforeEach(() => {
    fakeRenderComponent();
  });

  it("displays section name", async () => {
    expect(screen.getByTestId("section-name")).toHaveTextContent("Test");
  });

  it("displays email initial", async () => {
    expect(screen.getByTestId("email-initial")).toHaveTextContent("T");
  });

  it("matches snapshot", () => {
    const { container } = fakeRenderComponent();
    expect(container).toMatchSnapshot();
  });
});
