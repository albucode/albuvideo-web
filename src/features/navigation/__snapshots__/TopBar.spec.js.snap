// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar matches snapshot 1`] = `
<div>
  <div
    class="css-8q2c91"
  >
    <p
      class="chakra-text css-1dsg1cd"
      data-testid="section-name"
    >
      Test
    </p>
    <div
      class="css-gouzgf"
    >
      <div
        class="chakra-input__group css-9282n9"
      >
        <div
          class="chakra-input__left-element css-1n4mqg9"
        >
          <svg
            class="chakra-icon css-1godj66"
            focusable="false"
            viewBox="0 0 48 48"
          >
            <path
              d="M41.121 36.879L33.5055 29.2635C35.079 26.892 36 24.0525 36 21C36 12.729 29.271 6 21 6C12.729 6 6 12.729 6 21C6 29.271 12.729 36 21 36C24.0525 36 26.892 35.079 29.2635 33.5055L36.879 41.121C38.049 42.2925 39.951 42.2925 41.121 41.121C42.2925 39.9495 42.2925 38.0505 41.121 36.879ZM10.5 21C10.5 15.21 15.21 10.5 21 10.5C26.79 10.5 31.5 15.21 31.5 21C31.5 26.79 26.79 31.5 21 31.5C15.21 31.5 10.5 26.79 10.5 21Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <input
          class="chakra-input css-4cjkqg"
          placeholder="Search here..."
          type="text"
        />
      </div>
    </div>
    <button
      class="chakra-button css-1csw1ta"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-68klb0"
        focusable="false"
        viewBox="0 0 48 48"
      >
        <path
          d="M41.0401 30.42L37.4401 26.8V17.88C37.488 14.5677 36.3359 11.3499 34.1964 8.82071C32.0569 6.29156 29.0746 4.6219 25.8001 4.12003C23.8998 3.86977 21.9679 4.02766 20.1334 4.58318C18.2988 5.1387 16.6039 6.07908 15.1616 7.34156C13.7193 8.60403 12.5628 10.1596 11.7694 11.9044C10.9759 13.6493 10.5636 15.5433 10.5601 17.46V26.8L6.96015 30.42C6.50796 30.8798 6.20131 31.4626 6.07855 32.0956C5.95578 32.7287 6.02234 33.3839 6.2699 33.9793C6.51746 34.5747 6.93505 35.084 7.47045 35.4434C8.00585 35.8028 8.63531 35.9964 9.28015 36H16.0001V36.68C16.0936 38.7104 16.988 40.621 18.4876 41.9931C19.9871 43.3652 21.9695 44.0869 24.0001 44C26.0308 44.0869 28.0131 43.3652 29.5127 41.9931C31.0122 40.621 31.9067 38.7104 32.0001 36.68V36H38.7201C39.365 35.9964 39.9944 35.8028 40.5298 35.4434C41.0652 35.084 41.4828 34.5747 41.7304 33.9793C41.9779 33.3839 42.0445 32.7287 41.9217 32.0956C41.799 31.4626 41.4923 30.8798 41.0401 30.42ZM28.0001 36.68C27.8892 37.642 27.4114 38.5241 26.6663 39.1426C25.9211 39.7611 24.9661 40.0682 24.0001 40C23.0342 40.0682 22.0791 39.7611 21.334 39.1426C20.5889 38.5241 20.1111 37.642 20.0001 36.68V36H28.0001V36.68Z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="chakra-button css-1csw1ta"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-68klb0"
        focusable="false"
        viewBox="0 0 48 48"
      >
        <path
          d="M41.7675 20.1562H38.9612C38.5953 18.75 38.031 17.396 37.3042 16.1675L39.2946 14.1723C39.8981 13.5689 39.8981 12.5884 39.2946 11.985L36.0171 8.70637C35.4139 8.103 34.4355 8.10244 33.8323 8.70581L31.8324 10.6961C30.604 9.96928 29.25 9.40453 27.8438 9.03891V6.2325C27.8438 5.37928 27.1707 4.6875 26.3175 4.6875H21.6825C20.8293 4.6875 20.1562 5.37928 20.1562 6.2325V9.03891C18.75 9.40453 17.396 9.969 16.1675 10.6959L14.1723 8.70544C13.5689 8.10188 12.5884 8.10188 11.985 8.70544L8.70637 11.9829C8.103 12.5861 8.10244 13.5645 8.70581 14.1677L10.6961 16.1676C9.96909 17.396 9.40453 18.75 9.03891 20.1562H6.2325C5.37928 20.1562 4.6875 20.8293 4.6875 21.6825V26.3175C4.6875 27.1708 5.37928 27.8438 6.2325 27.8438H9.03891C9.40453 29.25 9.969 30.604 10.6959 31.8325L8.70525 33.8277C8.10188 34.4309 8.10188 35.4116 8.70525 36.0147L11.9827 39.2934C12.5861 39.897 13.5643 39.8976 14.1677 39.2941L16.1676 37.3039C17.396 38.0307 18.75 38.5953 20.1562 38.9612V41.7675C20.1562 42.6208 20.8293 43.3125 21.6825 43.3125H26.3175C27.1708 43.3125 27.8438 42.6207 27.8438 41.7675V38.9612C29.25 38.5953 30.604 38.031 31.8325 37.3042L33.8277 39.2946C34.4309 39.8981 35.4116 39.8981 36.0147 39.2946L39.2934 36.0171C39.897 35.4139 39.8976 34.4355 39.2941 33.8323L37.3039 31.8324C38.0307 30.604 38.5953 29.25 38.9612 27.8438H41.7675C42.6208 27.8438 43.3125 27.1707 43.3125 26.3175V21.6825C43.3125 20.8293 42.6207 20.1562 41.7675 20.1562ZM24 31.725C19.7335 31.725 16.275 28.2664 16.275 24C16.275 19.7335 19.7335 16.275 24 16.275C28.2665 16.275 31.725 19.7335 31.725 24C31.725 28.2665 28.2665 31.725 24 31.725Z"
          fill="currentColor"
        />
      </svg>
    </button>
    <span
      class="css-ksajp3"
    >
      <span
        class="css-5oxpfa"
        data-testid="email-initial"
      >
        T
      </span>
    </span>
  </div>
</div>
`;
